plugins/catalog-backend/src/ingestion/types.ts

```diff
@@ -68,3 +69,69 @@ export type ReadLocationError = {
   location: LocationSpec;
   error: Error;
 };
+
+//
+// LocationAnalyzer
+//
+
+export type LocationAnalyzer = {
+  generateConfig(
+    location: AnalyzeLocationRequest,
+  ): Promise<AnalyzeLocationResponse>;
+};
+
+export type AnalyzeLocationRequest = {
+  location: LocationSpec;
+};
+
+export type AnalyzeLocationResponse = {
+  existingEntityFiles: AnalyzeLocationExistingEntity[];
```

Comment:
Wonder if we could skip this part of the interface for now, since it's not used?

Reply:
These types were suggested by @freben so I pasted them all to have some reminder what was intended for the future for this feature. Of course I can remove them, but does it make any problem if they are left out here?

Reply:
I vote keep :) We talked before about the flow where you just wanna register a url and we go "oh there's already a file in there, let's make sure there's a location for it"